### PR TITLE
Config: Change syntax and updates all configs

### PR DESCRIPTION
**Pull request by mwestphal**
Change config file syntax to:

```
[
{
  "options": 
  {
    "filename": true,
    "camera-direction": "-1,-0.5,-1",
    "hdri-ambient": true,
    "translucency-support": true,
    "animation-progress": true
  }
},
{
  "match": ".*obj",
  "options": 
  {
    "axis": true,
    "tone-mapping": true
  }
}
]
```

also adds a test and updates doc

Later:

  Add config file control for interactions: https://github.com/f3d-app/f3d/pull/1740
  Add glob support in config files: Create issue
  Add complete interaction doc: Another PR
  Add command-script option: https://github.com/f3d-app/f3d/pull/1734
  Reenable test with command script: Another PR

### CODE REVIEW

**Review by Meakk — APPROVED** [review]
lgtm

File: application/F3DConfigFileTools.cxx [hunk 1]
```diff
@@ -177,12 +177,34 @@ F3DOptionsTools::OptionsEntries F3DConfigFileTools::ReadConfigFiles(const std::s
...
+    for (const auto& configBlock : json)
     {
-      // Add each config entry into an option dict
+      // Recover match if any
+      std::string match;
+      try
+      {
+        match = configBlock.at("match");
+      }
+      catch (nlohmann::json::exception&)
```
Comment (snoyer) [inline application/F3DConfigFileTools.cxx:189]:
```suggestion
      catch (nlohmann::json::out_of_range&)
```
more specific exception for expected missing key so it doesn't ignore actual issues unintentionally

File: application/F3DConfigFileTools.cxx [hunk 2]
```diff
@@ -177,12 +177,34 @@ F3DOptionsTools::OptionsEntries F3DConfigFileTools::ReadConfigFiles(const std::s
...
+        match = ".*";
+      }
+
+      // Recover options if any
+      nlohmann::ordered_json optionsBlock;
+      try
+      {
+        optionsBlock = configBlock.at("options");
+      }
+      catch (nlohmann::json::exception&)
```
Comment (snoyer) [inline application/F3DConfigFileTools.cxx:201]:
```suggestion
      catch (nlohmann::json::out_of_range&)
```
same as above

File: application/F3DConfigFileTools.cxx [hunk 3]
```diff
@@ -169,39 +169,79 @@ F3DOptionsTools::OptionsEntries F3DConfigFileTools::ReadConfigFiles(const std::s
...
+        {
+          optionsBlock = configBlock.at("options");
+        }
+        catch (nlohmann::json::out_of_range&)
         {
-          entry[item.key()] = nlohmann::to_string(item.value());
         }
-        else if (item.value().is_string())
+
+        if (optionsBlock.size() == 0)
```
Comment (snoyer) [inline application/F3DConfigFileTools.cxx:207]:
```suggestion
        if (optionsBlock.empty())
```
as is tradition

File: application/F3DConfigFileTools.cxx [hunk 4]
```diff
@@ -169,39 +169,79 @@ F3DOptionsTools::OptionsEntries F3DConfigFileTools::ReadConfigFiles(const std::s
...
+          for (const auto& item : optionsBlock.items())
+          {
+            if (item.value().is_number() || item.value().is_boolean())
+            {
+              entry[item.key()] = nlohmann::to_string(item.value());
+            }
+            else if (item.value().is_string())
+            {
+              entry[item.key()] = item.value().get<std::string>();
+            }
```
Comment (Meakk) [inline application/F3DConfigFileTools.cxx:226]: Is it working?
```suggestion
            if (item.value().is_number() || item.value().is_boolean() || item.value().is_string())
            {
              entry[item.key()] = nlohmann::to_string(item.value());
            }
```

Comment (mwestphal) [reply application/F3DConfigFileTools.cxx:226]: nope

**Pull request merged**